rpcs3/Emu/RSX/RSXThread.cpp

```diff
@@ -945,6 +975,21 @@ namespace rsx
 
 				// Update other sub-units
 				zcull_ctrl->update(this);
+
+				if (new_get_put != umax)
```

Comment:
Second check goes away if the producer doesn't also consume its own data.
I have some reservations about doing this here still, but atomic access in a loop isn't exactly free so its fine.
I'd prefer to have it in do_local_task though, there is a block that executes under safe FIFO conditions there.

Reply:
Done.